WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -38,21 +45,352 @@ class WordPressAuthenticatorTests: XCTestCase {
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy,
+                                                     xmlrpc: "https://example.com/xmlrpc.php",
+                                                     username: "username",
+                                                     connectedEmail: "email-address@example.com")
+        
+        guard let navController = presenterSpy.presentedVC as? LoginNavigationController, let controller = navController.viewControllers.first as? LoginEmailViewController else {
+            XCTFail("Could not fetch correct ViewController")
+            return
+        }
+        
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertEqual(controller.loginFields.restrictToWPCom, true)
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogXMLRPC, "https://example.com/xmlrpc.php")
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogUsername, "username")
+        XCTAssertEqual(controller.loginFields.username, "email-address@example.com")
+    }
+    
+    func testShowLoginForSelfHostedSitePresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+    
+    func testShowLoginForSelfHostedSiteTracksOpenLogin() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let delegateSpy = WordPressAuthenticatorDelegateSpy()
+        
+        WordPressAuthenticator.shared.delegate = delegateSpy
+        
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        let trackedEvent = delegateSpy.trackedElement
+        
+        XCTAssertEqual(trackedEvent, WPAnalyticsStat.openedLogin)
+    }
+    
+    func testSignInForWPComReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPCom()
+        
+        XCTAssertTrue((vc as Any) is LoginEmailViewController)
+    }
+    
+    func testSignInForWPComWithLoginFieldsReturnsVC() {
+        let navController = WordPressAuthenticator.signinForWPCom(dotcomEmailAddress: "example@email.com", dotcomUsername: "username") as! UINavigationController
+        let vc = navController.topViewController
+        
+        XCTAssertTrue((navController as Any) is UIViewController)
```

Comment:
I _think_ this assertion is redundant because [`UINavigationController` is a subclass of `UIViewController`](https://developer.apple.com/documentation/uikit/uinavigationcontroller) and the tests already ensures that the instance is a `UINavigationController`.

Reply:
Good call, removed that assertion